pwnlib/heap/bins/bin.py

```diff
@@ -0,0 +1,162 @@
+from pwnlib.heap.basic_formatter import BasicFormatter
+
+
+class Bins(object):
+    """Base class to be inherit by the bins sequences. This class provides
+    the methods to access the bins array as well as standard implementation
+    of the __str__ method.
+
+    Attributes:
+        bins (:obj:`list` of :class:`Bin`): The bins of the sequence.
+    """
+
+    def __init__(self, bins):
+        self._bins = bins
+        self._basic_formatter = BasicFormatter()
+
+    @property
+    def bins(self):
+        return self._bins
+
+    def __getitem__(self, item):
+        return self._bins[item]
+
+    def __len__(self):
+        return len(self._bins)
+
+    def __iter__(self):
+        return iter(self._bins)
+
+    def __str__(self):
+        msg = [
+            self._basic_formatter.header(self._name()),
+            self._format_bins(self._start_index()),
+            self._basic_formatter.footer()
+        ]
+        return "\n".join(msg)
+
+    def _name(self):
+        return self.__class__.__name__
+
+    def _start_index(self):
+        return 0
+
+    def _format_bins(self, start_index=0):
+        bins_str = []
+        for i, bin_ in enumerate(self.bins):
+            if len(bin_) > 0:
+                bins_str.append("[{}] {}".format(
+                    i + start_index, bin_)
+                )
+
+        if bins_str:
+            return "\n".join(bins_str)
+        else:
+            return "    [-] No chunks found"
+
+    def summary(self, start_index=0):
+        bins_str = []
+        for i, bin_ in enumerate(self.bins):
+            if len(bin_) > 0:
+                bins_str.append(
+                    "    [{}] {:#x} ({})".format(
+                        start_index + i, bin_.chunks_size, len(bin_))
+                )
+
+        if bins_str:
+            return "\n".join(bins_str)
+        else:
+            return "    [-] No chunks found"
+
+
+class Bin(object):
+    """Base class to be inherit by the bins. This class provides the basic info
+    of the bin entry as well as the chunks of the bin.
+
+    Attributes:
+        bin_entry (BinEntry): The entry of malloc_state or
```

Comment:
Instead of documenting synthetic properties this way, please just add a doctoring to the individual members, e.g.

```py
    @property
     def bin_entry(self):
         """The entry of malloc_state or ..."""
         return self._bin_entry